src/pip/_internal/collector.py

```diff
@@ -515,11 +520,16 @@ def collect_links(self, project_name):
             if self.session.is_secure_origin(link)
         ]
 
-        logger.debug('%d location(s) to search for versions of %s:',
-                     len(url_locations), project_name)
-
-        for location in url_locations:
-            logger.debug('* %s', location)
+        url_locations = _remove_duplicate_links(url_locations)
+        lines = [
+            '{} location(s) to search for versions of {}:'.format(
+                len(url_locations),
```

Comment:
I would put these arguments on the same line since they fit. They would also lines up more closely with the `{}` they are replacing in the string.